config/v1alpha1/types_cluster_monitoring.go

```diff
@@ -77,6 +79,13 @@ type ClusterMonitoringSpec struct {
 	// userDefined set the deployment mode for user-defined monitoring in addition to the default platform monitoring.
 	// +required
 	UserDefined UserDefinedMonitoring `json:"userDefined"`
+
+	// metricsServer defines the configuration for the Metrics Server component.
+	// The Metrics Server provides container resource metrics for use in autoscaling pipelines.
+	// When omitted, this means no opinion and the platform is left to choose a default,
+	// which is subject to change over time.
+	// +optional
```

Comment:
Explicit mention that the field is optional in the GoDoc. "When omitted" implies that, but I generally prefer being explicit.